context/src/main/java/io/opentelemetry/context/propagation/CompositeTextMapPropagatorBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context.propagation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A builder for configuring an {@link TextMapPropagator} specifying which propagators should be used for extracting the context and which should be used for injecting the context.
+ */
+public final class CompositeTextMapPropagatorBuilder {
+
+  private final List<TextMapPropagator> extractors;
+  private final List<TextMapPropagator> injectors;
+
+  /**
+   * Package protected to disallow direct initialization.
+   *
+   * @see TextMapPropagator#builder()
+   */
+  CompositeTextMapPropagatorBuilder() {
+    this.extractors = new ArrayList<>();
+    this.injectors = new ArrayList<>();
+  }
+
+  /**
+   * Add a {@link TextMapPropagator} to be used only to extract the context.
+   */
+  public CompositeTextMapPropagatorBuilder extractor(TextMapPropagator propagator) {
+    this.extractors.add(propagator);
+    return this;
+  }
+
+  /**
+   * Add a {@link TextMapPropagator} to be used only to inject the context.
+   */
+  public CompositeTextMapPropagatorBuilder injector(TextMapPropagator propagator) {
+    this.injectors.add(propagator);
+    return this;
+  }
+
+  /**
+   * Add a {@link TextMapPropagator} to be used both to extract and inject the context.
+   */
+  public CompositeTextMapPropagatorBuilder propagator(TextMapPropagator propagator) {
+    this.injectors.add(propagator);
+    this.extractors.add(propagator);
+    return this;
+  }
+
+  /**
+   * Returns the built {@link TextMapPropagator}
+   *
+   * @see CompositeTextMapPropagatorBuilder
+   */
+  public TextMapPropagator build() {
+    if (this.injectors.isEmpty()) {
+      this.injectors.add(TextMapPropagator.noop());
```

Comment:
We don't need these blocks right?

Reply:
You mean we don't need noop injectors and extractor? I see that `TextMapPropagator#composite` adds them

Reply:
`TextMapPropagator#composite` replaces the `MultiTextMapPropagator` completely with no-op if there aren't any propagators. So yeah I guess we can do similar here, it is

```java
if (injectors.isEmpty() && extractors.isEmpty()) {
  return NoopTextMapPropagator.getInstance();
}
return new MultiTextMapPropagator(injectors, extractors);
```

Reply:
Makes sense!